### DISCUSSION TITLE
is a Photo Interrupter Sensor included in tasmota

### DISCUSSION THREAD

**Discussion by montyfert** [arendst/Tasmota#20080] [Q&A]
hi i cant seem to fnd any info about tasmota and the Photo Interrupter Sensor  i have a d1mini with tasmota sensors installed ,
![image_2023-11-21_163441158](https://github.com/arendst/Tasmota/assets/74438819/eb94f9b9-d41f-4217-bf32-2536f4815e6e)
im trying to make a way to measure filament length , 
any help would be much apreciated .
thanks 
martin

**Comment by sfromis** [comment]
That does not look like needing device-specific support, but just act as a standard switch input, where there is a number of config options for how to handle that.

Not sure what you mean by "measure filament length", but if it is simply something present or not, that should work as a simple Switch input config.

**Comment by montyfert** [comment]
hi sorry shoud have explained better, i want to measure the filament use of my 3d printer so the filament will be turning a wheel with holes in it imbetween the sensor i want to be able to measure the length based on the wheel turning .

**Reply by sfromis** [reply]
Still not clear what "measure" means, but it sounds like counting pulses. And sure, Tasmota should be able to detect the pulses.

**Reply by montyfert** [reply]
measure the 
length in mm

**Reply by sfromis** [reply]
Does it mean Tasmota doing anything more than counting pulses from that sensor?

**Reply by montyfert** [reply]
at the moment yes , then i have to sort a rule out to work out how many pulses equals 10mm etc . thanks
martin

**Reply by sfromis** [reply]
Tasmota can register pulses no problem, assuming that the electrical and wiring setup works. You can get messages per pulse, or you can set up a counter incrementing by pulse.

**Reply by montyfert** [reply]
the counter incrementing by pulse sounds about right but is it posible to backwards as well decrese

**Reply by sfromis** [reply]
Yes, you can reset the built-in counter feature to any number, or you can add a negative number.

**Reply by montyfert** [reply]
thats great

**Reply by montyfert** [reply]
thanks 
martin
